projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -679,6 +763,13 @@ export class IgxTreeGridComponent extends IgxGridBaseDirective implements GridTy
         }
         this.selectionService.clearHeaderCBState();
         this._pipeTrigger++;
+        if (this.rowSelection === 'multipleCascade') {
+            this.cdr.detectChanges();
```

Comment:
I don't get what's the change to detect here, plus, why run it before checking if the parent is selected anyway? And then considering children are available right here, could they not just be added to the selection directly?